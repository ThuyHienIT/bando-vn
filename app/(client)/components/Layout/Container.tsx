'use client';

import { Layout } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';

const ContainerStyle = styled.div`
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
`;

interface Props {
  children: React.ReactNode;
  className?: string;
}
export const Container = memo<Props>(function Header(props) {
  return <ContainerStyle className={props.className}>{props.children}</ContainerStyle>;
});
