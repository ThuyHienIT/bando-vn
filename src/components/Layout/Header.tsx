import { Layout, Typography } from 'antd';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { memo, useCallback } from 'react';
import styled from 'styled-components';

const HeaderStyle = styled(Layout.Header)`
  position: sticky;
  top: 0;
  z-index: 1;
  width: 100%;
  background-color: #f0f2f5;
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid #dddddd;
`;

const LogoStyle = styled(Link)`
  height: 31px;
  margin: 16px 24px 16px 0;
  background: rgba(255, 255, 255, 0.2);
  display: inline-flex;
  align-items: center;
  gap: 8px;
`;

const Nav = styled.div`
  flex-grow: 1;
`;

interface Props {}

export const Header = memo<Props>(function Header(props) {
  const router = useRouter();
  console.log('router', router);

  const handleClick = useCallback(
    ({ key }: { key: string }) => {
      router.push(`/admin/${key}`);
    },
    [router]
  );

  return (
    <HeaderStyle>
      <LogoStyle href="/admin">
        <img src="/logo.png" alt="Facility Booking" height={30} />
        <Typography.Title level={4} style={{ margin: 0 }}>
          VN Map - Admin
        </Typography.Title>
      </LogoStyle>
    </HeaderStyle>
  );
});
