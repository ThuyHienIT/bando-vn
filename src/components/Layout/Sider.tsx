import { Avatar, Button, Layout, Menu, Typography } from 'antd';
import Link from 'next/link';
import router, { useRouter } from 'next/router';
import { memo, useCallback, useMemo, useState } from 'react';
import { useRecoilValue } from 'recoil';
import styled from 'styled-components';

import {
  ArrowLeftOutlined,
  ArrowRightOutlined,
  BankOutlined,
  CoffeeOutlined,
  DollarCircleOutlined,
  EnvironmentOutlined,
  PictureOutlined,
  ShoppingCartOutlined,
  ToolOutlined,
  UserOutlined
} from '@ant-design/icons';
import { userInfoState } from '@recoil/user';

const SidebarStyle = styled(Layout.Sider)`
  background-color: #f0f2f5;
  border-right: 1px solid #dddd;
  overflow: auto;
  height: 100vh;
  position: sticky;
  top: 0;

  .ant-layout-sider-children {
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
`;
const UserProfileStyle = styled.div`
  position: absolute;
  top: 16px;
  left: 16px;
  display: flex;
  gap: 8px;
  align-items: center;

  .ant-avatar {
    flex-shrink: 0;
  }
`;

const CollapseBtn = styled(Button)`
  position: absolute;
  bottom: 16px;
  left: 16px;
`;

const MenuStyle = styled(Menu)`
  background-color: transparent;

  .ant-menu-item {
    margin: 8px;
  }

  &.ant-menu:not(.ant-menu-horizontal) .ant-menu-item-selected {
    background-color: #ffffff;
    border-radius: 8px;
  }

  &.ant-menu.ant-menu-inline-collapsed > .ant-menu-item {
    padding: 0 calc(50% - 34px / 2);
  }
`;

interface Props {}
export const Sider = memo<Props>(function Sider(props) {
  const router = useRouter();
  const [collapsed, setCollapsed] = useState(false);
  const userInfo = useRecoilValue(userInfoState);

  const handleCollapse = useCallback(() => {
    setCollapsed((c) => !c);
  }, []);

  const type = useMemo(() => {
    if (router.query.type) return router.query.type as string;

    return router.pathname.replace('/admin/', '').split('/')[0];
  }, [router]);

  return (
    <SidebarStyle
      width={200}
      className="site-layout-background"
      theme="light"
      collapsed={collapsed}
      collapsible
      trigger={null}
    >
      <UserProfileStyle>
        <Avatar size="large" src="https://picsum.photos/200/200" />
        {!collapsed && (
          <Typography.Title
            style={{ margin: 0, whiteSpace: 'nowrap' }}
            level={5}
          >
            {userInfo?.name}
          </Typography.Title>
        )}
      </UserProfileStyle>

      <MenuStyle
        mode="vertical"
        theme="light"
        selectedKeys={[type]}
        items={[
          {
            icon: <CoffeeOutlined />,
            label: <Link href="/admin/restaurant">Restaurant</Link>,
            key: 'restaurant',
          },
          {
            icon: <ShoppingCartOutlined />,
            label: <Link href="/admin/shopping">ShoppingMall</Link>,
            key: 'shopping',
          },
          {
            icon: <BankOutlined />,
            label: <Link href="/admin/hotel">Hotel</Link>,
            key: 'hotel',
          },
          {
            icon: <EnvironmentOutlined />,
            label: <Link href="/admin/attractions">Attractions</Link>,
            key: 'attraction',
          },
          {
            icon: <DollarCircleOutlined />,
            label: <Link href="/admin/agency">Agency</Link>,
            key: 'agency',
          },
          {
            icon: <PictureOutlined />,
            label: <Link href="/admin/tour">Tour</Link>,
            key: 'tour',
          },
          {
            icon: <ToolOutlined />,
            label: <Link href="/admin/settings">Settings</Link>,
            key: 'settings',
          },
          ...(userInfo?.role === 'admin'
            ? [
                {
                  icon: <UserOutlined />,
                  label: <Link href="/admin/user">Users</Link>,
                  key: 'user',
                },
              ]
            : []),
        ]}
      />

      <CollapseBtn
        onClick={handleCollapse}
        icon={collapsed ? <ArrowRightOutlined /> : <ArrowLeftOutlined />}
      ></CollapseBtn>
    </SidebarStyle>
  );
});
