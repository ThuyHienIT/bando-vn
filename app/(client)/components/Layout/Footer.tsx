'use client';

import { Layout, Typography } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';
import { Container } from './Container';

interface Props {}
export const Footer = memo<Props>(function Header(props) {
  return (
    <Layout.Footer style={{ padding: '8px 0' }}>
      <Container>
        <Typography.Text type="secondary">
          <small>Â© Created by Thang Kieu</small>
        </Typography.Text>
      </Container>
    </Layout.Footer>
  );
});
