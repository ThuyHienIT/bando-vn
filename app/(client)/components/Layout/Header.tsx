'use client';

import { ProfileOutlined } from '@ant-design/icons';
import { Layout, Typography } from 'antd';
import Image from 'next/image';
import { memo } from 'react';
import styled from 'styled-components';
import { UserProfile } from '../UserProfile';
import { Container } from './Container';

const HeaderStyle = styled(Layout.Header)`
  &&& {
    background-color: #fff;
  }
`;

const HeaderContent = styled(Container)`
  height: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const Logo = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;

  .ant-typography {
    margin-bottom: 0;
  }
`;

interface Props {}
export const Header = memo<Props>(function Header(props) {
  return (
    <HeaderStyle>
      <HeaderContent>
        <Logo>
          <Image src="/logo.png" alt="Facility Booking" width={30} height={30} />
          <Typography.Title level={4}>Facility Booking</Typography.Title>
        </Logo>
        <UserProfile />
      </HeaderContent>
    </HeaderStyle>
  );
});
