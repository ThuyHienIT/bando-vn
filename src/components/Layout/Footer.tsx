import { Layout, Typography } from 'antd';
import { memo } from 'react';
import styled from 'styled-components';

import { Container } from './Container';

const FooterStyle = styled(Layout.Footer)`
  padding: 8px 0;
  border-top: 2px solid #dddddd;
`;

interface Props {}
export const Footer = memo<Props>(function Header(props) {
  return (
    <FooterStyle style={{ padding: '8px 0' }}>
      <Container>
        <Typography.Text type="secondary">
          <small>Â© Created by Thang Kieu</small>
        </Typography.Text>
      </Container>
    </FooterStyle>
  );
});
