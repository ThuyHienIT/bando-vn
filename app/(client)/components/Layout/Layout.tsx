'use client';

import { Layout } from 'antd';
import { memo } from 'react';
import { Header } from './Header';
import styled from 'styled-components';
import { Footer } from './Footer';
import { Container } from './Container';

const Content = styled(Layout.Content)``;

interface Props {
  children: React.ReactNode;
}
export const BasicLayout = memo<Props>(function BasicLayout(props) {
  return (
    <Layout>
      <Header />
      <Content className="site-layout">
        <Container>{props.children}</Container>
      </Content>
      <Footer />
    </Layout>
  );
});
