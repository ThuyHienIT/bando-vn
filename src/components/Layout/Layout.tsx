import { Layout } from 'antd';
import { memo, useEffect } from 'react';
import { useRecoilState } from 'recoil';
import styled from 'styled-components';

import { userInfoState } from '@recoil/user';

import { Container } from './Container';
import { Footer } from './Footer';
import { Header } from './Header';

const LayoutStyle = styled(Layout)`
  &&& {
    min-height: 100vh;
  }
`;

interface Props {
  children: React.ReactNode;
  userInfo?: UserInfo;
}
export const BasicLayout = memo<Props>(function BasicLayout(props) {
  const [userInfo, updateUserInfo] = useRecoilState(userInfoState);

  useEffect(() => {
    if (!userInfo && props.userInfo) {
      updateUserInfo(props.userInfo);
    }
  }, [props.userInfo]);

  return (
    <LayoutStyle>
      <Header />
      <Layout.Content>
        <Container className="main-container">{props.children}</Container>
      </Layout.Content>
      <Footer />
    </LayoutStyle>
  );
});
