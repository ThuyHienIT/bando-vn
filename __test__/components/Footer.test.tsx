import { render, screen } from '@testing-library/react';
// import Home from '../pages/index'
import '@testing-library/jest-dom';
import { Footer } from '@components/Layout/Footer';

describe('Footer', () => {
  it('renders footer unchanged', () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
