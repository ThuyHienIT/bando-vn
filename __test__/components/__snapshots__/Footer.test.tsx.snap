// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders footer unchanged 1`] = `
.c1 {
  max-width: 1080px;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  width: 100%;
}

.c0 {
  padding: 8px 0;
  border-top: 2px solid #dddddd;
}

<div>
  <footer
    class="ant-layout-footer c0"
    style="padding: 8px 0px;"
  >
    <div
      class="c1"
    >
      <span
        class="ant-typography ant-typography-secondary css-dev-only-do-not-override-diro6f"
      >
        <small>
          Â© Created by Thang Kieu
        </small>
      </span>
    </div>
  </footer>
</div>
`;
